Microsoft.Toolkit.Uwp/Extensions/StringExtensions.cs

```diff
@@ -0,0 +1,57 @@
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Net;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Windows.ApplicationModel.Resources;
+
+namespace Microsoft.Toolkit.Uwp.Extensions
+{
+    /// <summary>
+    /// UWP specific helpers for working with strings.
+    /// </summary>
+    public static class StringExtensions
+    {
+
+        private static ResourceLoader _independentLoader;
+
+
+        /// <summary>
+        /// Retrieves the provided resource for the current view context.
+        /// </summary>
+        /// <param name="resourceKey">Resource key to retrieve.</param>
+        /// <returns>string value for given resource or empty string if not found.</returns>
+        public static string GetViewLocalized(this string resourceKey)
+        {
+
+            _independentLoader = ResourceLoader.GetForViewIndependentUse();
+            return ResourceLoader.GetForCurrentView().GetString(resourceKey);
+        }
+
+        /// <summary>
+        /// Retrieves the provided resource for the given key for use independent of the UI thread.
+        /// </summary>
+        /// <param name="resourceKey">Resource key to retrieve.</param>
+        /// <returns>string value for given resource or empty string if not found.</returns>
+        public static string GetLocalized(this string resourceKey)
+        {
+            _independentLoader = ResourceLoader.GetForViewIndependentUse();
+            return _independentLoader.GetString(resourceKey);
+        }
+
+        /// <summary>
+        /// Retrieves the provided resource for the given key for use independent of the UI thread.
+        /// </summary>
+        /// /// <param name="resourcePath">Resource path to retrieve.</param>
+        /// <param name="resourceKey">Resource key to retrieve.</param>
+        /// <returns>string value for given resource or empty string if not found.</returns>
+        public static string GetLocalized(string resourcePath, string resourceKey)
```

Comment:
Why this extra method?  You can pass the full path into the method above and it should behave the same, that's why I had tested before.  Otherwise, it's an odd place for this here, as it's not an extension method.  If you think this is important still, I'd expect signature to be `public static string GetLocalized(this string resourceKey, string resourcePath)`.  Then construct a non-cached resource loader `return ResourceLoader.GetForViewIndependentUse(resourcePath).GetString(resourceKey);`.

Reply:
We did this because the current getLocalized method doesn't retrieve our ResourceMap. So, we created an extra method to get it by passing the resource path as a parameter.

Reply:
I tested this out on my branch before suggesting it, so I'm not sure what's different.  I'll have to compare when I'm back in the office tomorrow.  This method could still have use, but we should write it as an extension still:

```suggestion
        public static string GetLocalized(this string resourceKey, string resourcePath)
```

And then leave the `_independentLoader` initialization to the static value.  The point of that one is to cache it for repeated calls to `GetLocalized`.